backend/src/v5/middleware/dataConverter/inputs/users.js

```diff
@@ -169,6 +171,90 @@ Users.validateResetPasswordData = async (req, res, next) => {
 	}
 };
 
+Users.validateSignUpData = async (req, res, next) => {
+	const schema = Yup.object().shape({
+		username: types.strings.username.test('checkUsernameAvailable', 'Username already exists',
+			async (value) => {
+				if (value) {
+					try {
+						await getUserByUsername(value, { _id: 1 });
+						return false;
+					} catch {
+					// do nothing
+					}
+				}
+				return true;
+			}).required(),
+		email: types.strings.email.test('checkEmailAvailable', 'Email already exists',
+			async (value) => {
+				if (value) {
+					try {
+						await getUserByQuery({ 'customData.email': value }, { _id: 1 });
+						return false;
+					} catch {
+						// do nothing
+					}
+				}
+				return true;
+			}).required(),
+		password: types.strings.password.required(),
+		firstName: types.strings.name.required(),
+		lastName: types.strings.name.required(),
+		countryCode: types.strings.countryCode.required(),
+		company: Yup.string().max(120).optional(),
```

Comment:
the reason I did that is because even if I make it optional it rejects an empty string (has to have min 1 chars) and the form passes empty string if you dont populate company

Reply:
that's technically correct, you either don't pass it in or you pass in something valid, the intention should be clear.

If we take in empty strings, we need to make sure we're not storing empty strings, we should strip them.